trl/trainer/grpo_trainer.py

```diff
@@ -752,7 +864,30 @@ def _set_signature_columns_if_needed(self):
         # In GRPOTrainer, we preprocess data, so using the model's signature columns doesn't work.
         # Instead, we set them to the columns expected by the `training_step` method, hence the override.
         if self._signature_columns is None:
-            self._signature_columns = ["prompt"]
+            # Base signature columns
+            signature_columns = ["prompt"]
+
+            if self.is_vision_model:
+                vlm_columns = ["image", "pixel_values", "pixel_attention_mask", "image_sizes"]
+                signature_columns.extend(vlm_columns)
+
+            # Additional columns needed based on the reward functions
+            for func in self.reward_funcs:
+                if not isinstance(func, (str, PreTrainedModel)):
+                    try:
+                        # Custom reward function - inspect its signature to see what columns it expects
+                        func_sig = inspect.signature(func)
+                        params = func_sig.parameters
+                        # Add any parameter that's not one of the expected defaults
+                        for param_name in params:
+                            if param_name not in ["prompts", "completions", "kwargs", "trainer_state"]:
+                                if param_name not in signature_columns:
+                                    signature_columns.append(param_name)
+                    except (ValueError, TypeError):
+                        # If we can't inspect the function, just skip it
+                        pass
```

Comment:
I don't get why all of this is needed. I'll try with something simpler, and revert if I some point it makes sense for me

Reply:
we dont need it, my thinking was that  there might be some extra columns in the data for vlms that might need passing to the reward model but you are right we don't need this and I'll remove it